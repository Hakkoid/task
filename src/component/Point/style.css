.Point{
	padding: 5px;
	margin: 1px 0px;
	color: #333333;
	background: #ffffff;
	min-height: 30px;
}


.Point_focus{
	transition: box-shadow 3s line 2s;
	box-shadow: 0px 10px 50px 5px rgba(0, 0, 0, 0.25);
	animation: Point_shadow 400ms 1 ease;
}

@keyframes Point_shadow{
	from{
		box-shadow: 0px 0px 0px 0px rgba(0, 0, 0, 0);
	}
	to{
		box-shadow: 0px 10px 50px 5px rgba(0, 0, 0, 0.25);
	}
}

.Point__Text{
	padding: 0;
	margin: 0;
	width: calc(100% - 25px);
	font-size: 16px;
	overflow: hidden;
	display: inline-block;
}

.Point__Description{
	padding: 0;
	margin: 0;
	font-size: 12px;
	width: calc(100% - 25px);
	overflow: hidden;
	display: inline-block;
}

.Point__Close{
	background: #ffffff;
	border: none;
	float: right;
	padding: 2px;
}

.Point__Close>span {
	position: relative;
	width: 16px;
	height: 16px;
	opacity: 0.5;
	float: right;
}

.Point__Close>span:hover {
	opacity: 1;
}

.Point__Close>span:before, .Point__Close>span:after {
	position: absolute;
	top: 2px;
	left: 7px;
	content: ' ';
	height: 12px;
	width: 2px;
	background-color: #333333;
}

.Point__Close>span:before {
	transform: rotate(45deg);
}

.Point__Close>span:after {
	transform: rotate(-45deg);
}