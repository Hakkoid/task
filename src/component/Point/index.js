import React, { Component } from 'react';
import PropTypes from 'prop-types'
import './style.css';

class Point extends Component {
  constructor(props){
    super(props);
    this.handleClose = this.handleClose.bind(this);
  }

  handleClose(e){
    this.props.onClose(this.props.id);
  }

  render() {
    return (
      <li className="Point">
        <p className="Point__Text">{this.props.children}</p>
        <button onClick={this.handleClose} className="Point__Close"><span /></button>
      </li>
    );
  }
}

Point.propTypes = {
  onClose: PropTypes.func.isRequired,
  id: PropTypes.string.isRequired
}

export default Point;