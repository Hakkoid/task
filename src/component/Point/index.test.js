import React from 'react';
import { mount } from 'enzyme';
import Point from './index';
import renderer from 'react-test-renderer';


describe( 'Point', () => {

    it('renders without crashing', () => {
        mount(<Point id = '0' onClose = {() => {}} />)
    })


    describe( 'onClose event', () => {
        
        it('should send the id prop to the callback func', () => {

            let handler = (event) => { expect(event).toBe(id)}

            let id = '0';
            let component = mount(<Point id = {id} onClose = {handler}/>);
            component.find('.Point__Close').simulate('click')

            id = 'ssss';
            component.setProps({id})
            component.find('.Point__Close').simulate('click')
        })

    })


    it( 'Snapshot', () => {
        const component = renderer.create(
        <Point id = '0' onClose = { () => {}} />,
        );

        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    })
})