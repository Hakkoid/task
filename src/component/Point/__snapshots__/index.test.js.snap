// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Point Snapshot 1`] = `
<li
  className="Point"
>
  <p
    className="Point__Text"
  />
  <button
    className="Point__Close"
    onClick={[Function]}
  >
    <span />
  </button>
</li>
`;
